acapy_plugin_pickup/protocol/delivery.py

```diff
@@ -153,76 +153,104 @@ async def remove_message_for_key(self, key: str, msg: OutboundMessage):
         """Remove specified message from queue for key."""
 
 
-class InMemoryQueue(UndeliveredInterface):
-    """In-memory undelivered message queue."""
+class PersistedQueue(UndeliveredInterface):
```

Comment:
`PersistedQueue` should be something like `RedisUndeliveredQueue`. `PersistedQueue` implies that it's a generic/ABC for all Queues providing persistence.